src/file_pipe_log/log_file.rs

```diff
@@ -117,10 +119,10 @@ impl<F: FileSystem> LogFileWriter<F> {
         Ok(())
     }
 
-    pub fn sync(&mut self) -> IoResult<()> {
+    pub fn sync(&mut self) {
```

Comment:
I think there's no need to change signature even if it never raises error. Especially for trait method `PipeLog::sync`.

Reply:
If it return a `Result`, wouldn't it confuse people of how should they deal with the error, like whether to yield the error up or just `unwrap()`?

I think leave it this way ease the future maintenance work?

Reply:
The characteristics of fsync (fsync syscall, not sync method here) is file-system dependent. It should not affect how user interface is designed. For example, we can do internal buffering inside writer, then the pwrite syscall before fsync syscall will not trigger a panic. As for "how should they deal with the error", you just need to add some comments to explain the guarantee of the function (both the trait method and writer method).

Reply:
Done